### ISSUE TITLE
[explicit-member-accessibility] parameterProperties: 'explicit' not working as expected

### ISSUE THREAD

**Issue by ckknight** [typescript-eslint/typescript-eslint#857]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-member-accessibility": ["error", {
      "accessibility": "explicit"
    }]
  }
}
```

or

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-member-accessibility": ["error", {
      "accessibility": "off",
      "overrides": { "parameterProperties": "explicit" }
    }]
  }
}
```

```TS
export class WithParameterProperty {
  public constructor(readonly value: string) {}
}
```

**Expected Result**
An error that the parameter property `value` needs an accessibility modifier like `private`, `protected`, or `public`

**Actual Result**
No error

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0` |
| `@typescript-eslint/parser`        | `2.0.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `12.6.0` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
See #631
